libs/coin-modules/coin-stellar/src/polyfill.ts

```diff
@@ -0,0 +1,47 @@
+import { log } from "@ledgerhq/logs";
+
+/*
+ * This whole patch thing is messy:
+ * It's goal is to: Fix error when sending a token transaction in Stellar
+ * It originally was in ledger-mobile polyfill.ts (https://github.com/LedgerHQ/ledger-live/pull/9687)
+ * But globally patching is not a good idea, and broke the polkadot sdk
+ */
+
+// Save original methods once at app startup
+const OriginalTypedArrayMethods = {
+  subarray: Uint8Array.prototype.subarray,
+  slice: Uint8Array.prototype.slice,
+  map: Uint8Array.prototype.map,
+  filter: Uint8Array.prototype.filter,
+};
+
+// Track whether we actually patched
+let hermesTypedArrayPatched = false;
+
+export function patchHermesTypedArraysIfNeeded() {
+  if (global.HermesInternal) {
+    if (!hermesTypedArrayPatched) {
```

Comment:
```suggestion
  if (global.HermesInternal && !hermesTypedArrayPatched) {
```

Also we may want to negate the expression and go for an early return ... Reading preferences only, props to you